app/controllers/users_controller.rb

```diff
@@ -1,28 +1,84 @@
 class UsersController < ApplicationController
+  before_action :logged_in_user, except: %i(index edit update destroy)
+  before_action :load_users, only: %i(edit show update delete)
+  before_action :correct_user, only: %i(edit update)
+  before_action :admin_user, except: %i(destroy)
+
   def new
     @user = User.new
   end
 
   def show
```

Comment:
```suggestion
  def show; end
```
để method show thành như này kia mà, sao mãi k sửa nhỉ

Reply:
method show của em có nội dung mà a ơi :((
def show
    load_users
    return if @user
      flash[:error] = t (".notfound")
      redirect_to root_path
  end